services/src/unit-test/java/uk/gov/moj/sdt/services/messaging/MessageWriterTest.java

```diff
@@ -110,87 +111,73 @@ public void testQueueMessageWithEmptyTargetApp() {
         // Setup finished, now tell the mock what to expect.
         final ISdtMessage sdtMessage = new SdtMessage();
         sdtMessage.setSdtRequestReference("Test");
 
-        messageSender.sendMessage("UnitTestQueue", sdtMessage);
-        EasyMock.expectLastCall();
-
-        // Get ready to call the mock.
-        replay(messageSender);
+        messageSender.sendMessage(UNIT_TEST_QUEUE, sdtMessage);
 
         // Send the message.
         try {
             messageWriter.queueMessage(sdtMessage, "UnitTest", false);
-            Assert.fail("Should have thrown an Illegal Argument exception here as the target app code is invalid");
+            fail("Should have thrown an Illegal Argument exception here as the target app code is invalid");
         } catch (final IllegalArgumentException e) {
-            Assert.assertTrue("Target application code does not have a mapped queue name", true);
+            assertTrue(true, "Target application code does not have a mapped queue name");
         }
     }
 
     /**
      * Test for a valid queue message on the MessageWriter.
      */
     @Test
-    public void testQueueMessage() {
+    void testQueueMessage() {
         // Setup finished, now tell the mock what to expect.
         final ISdtMessage sdtMessage = new SdtMessage();
         sdtMessage.setSdtRequestReference("Test");
 
-        messageSender.sendMessage("UnitTestQueue", sdtMessage);
-        EasyMock.expectLastCall();
-
-        // Get ready to call the mock.
-        replay(messageSender);
+        messageSender.sendMessage(UNIT_TEST_QUEUE, sdtMessage);
 
         // Send the message.
         try {
-            final Map<String, String> queueNameMap = new HashMap<String, String>();
-            queueNameMap.put("UNITTEST", "UnitTestQueue");
+            final Map<String, String> queueNameMap = new HashMap<>();
+            queueNameMap.put(UNIT_TEST, UNIT_TEST_QUEUE);
 
             messageWriter.setQueueNameMap(queueNameMap);
 
-            messageWriter.queueMessage(sdtMessage, "UNITTEST", false);
-            Assert.assertTrue("Success", true);
+            messageWriter.queueMessage(sdtMessage, UNIT_TEST, false);
+            assertTrue(true, "Success");
         } catch (final IllegalArgumentException e) {
-            LOGGER.error("Error", e);
-            Assert.fail("Not Expected to fail");
+            fail("Not Expected to fail");
         }
 
-        EasyMock.verify(messageSender);
+        verify(messageSender, times(2)).sendMessage(any(), any());
     }
 
     /**
      * Test for a valid queue message on the MessageWriter for dead letter.
      */
     @Test
-    public void testQueueMessageForDeadLetter() {
+    void testQueueMessageForDeadLetter() {
         // Setup finished, now tell the mock what to expect.
         final ISdtMessage sdtMessage = new SdtMessage();
         sdtMessage.setSdtRequestReference("Test");
 
         messageSender.sendMessage("UnitTestQueue.DLQ", sdtMessage);
```

Comment:
This line should be setting up the expected behaviour on the mocked messageSender, not invoking it. This needs to be wrapped in either the Mockito doNothing() or doAnswer() methods and the verify() line below changed to expect only a single invocation instead of 2.

Reply:
The messageWriter.getQueueName() method appends the dead letter queue suffix to the queue name for this test, so the verification line should check that messageSender.sendMessage() is invoked with the amended queue name (UnitTetsQueue.DLQ) instead of using any().

Reply:
changed it now

Reply:
The last comment only referred to the testQueueMessageForDeadLetter() test as that's the only test that tries to add something to the dead letter queue (the final boolean parameter of the messageWriter.queueMessage() method controls whether the message is sent to the dead letter queue or not).

The tests should verify messageSender.sendMessage() as follows:

testQueueMessageWithEmptyTargetApp() and testQueueMessageWithInvalidTarget():
verify(messageSender, times(0)).sendMessage(UNIT_TEST_QUEUE, sdtMessage);

testQueueMessage():
verify(messageSender, times(1)).sendMessage(UNIT_TEST_QUEUE, sdtMessage);

testQueueMessageForDeadLetter():
verify(messageSender, times(1)).sendMessage("UnitTestQueue.DLQ", sdtMessage);

Might be worth setting up a constant for UnitTestQueue.DLQ - it'll be UNIT_TEST_QUEUE + ".DLQ"

Reply:
Confirmed test verification changes have been made.